app/models/manageiq/providers/redhat/inventory/collector/target_collection.rb

```diff
@@ -173,7 +173,7 @@ def infer_related_vm_ems_refs_db!
 
     changed_vms.each do |vm|
       add_simple_target!(:ems_clusters, vm.ems_cluster.ems_ref)
-      vm.storages.collect(&:ems_ref).compact.each { |ems_ref| add_simple_target!(:storagedomains, ems_ref) }
+      vm.storages.collect(&:ems_ref).compact.each { |ems_ref| add_simple_target!(:storagedomains, ems_ref) } unless vm.storages.empty?
```

Comment:
I think this should be vm.storages.nil?
because if its empty ([]) this code will not do anything anyway, while if its nil - nil.empty? throws an exception.